src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AssemblyAlignmentParser.java

```diff
@@ -0,0 +1,174 @@
+     * @param ctx                       spark context for IO operations
+     * @param pathToInputAlignments     path string to alignments of the contigs; format assumed to be consistent/parsable by {@link AlignmentRegion#toString()}
+     * @return                          an PairRDD for all assembled contigs with their alignment regions and sequence
+     */
+    static JavaPairRDD<Iterable<AlignmentRegion>, byte[]> prepAlignmentRegionsForCalling(final JavaSparkContext ctx,
+                                                                                         final String pathToInputAlignments,
+                                                                                         final String pathToInputAssemblies) {
+
+        final JavaPairRDD<Tuple2<String, String>, Iterable<AlignmentRegion>> alignmentRegionsKeyedByAssemblyAndContigId
+                = parseAlignments(ctx, pathToInputAlignments);
+
+        final JavaPairRDD<Tuple2<String, String>, byte[]> contigSequences
+                = loadContigSequence(ctx, pathToInputAssemblies);
+
+        return alignmentRegionsKeyedByAssemblyAndContigId
+                .join(contigSequences)
+                .mapToPair(Tuple2::_2);
+    }
+
+    // TODO: 11/23/16 test
+    static JavaPairRDD<Tuple2<String, String>, Iterable<AlignmentRegion>> parseAlignments(final JavaSparkContext ctx,
+                                                                                          final String pathToInputAlignments) {
+
+        return ctx.textFile(pathToInputAlignments).map(textLine -> AlignmentRegion.fromString(textLine.split(AlignmentRegion.STRING_REP_SEPARATOR,-1)))
+                .flatMap(oneRegion -> breakGappedAlignment(oneRegion, SVConstants.GAPPED_ALIGNMENT_BREAK_DEFAULT_SENSITIVITY))
+                .mapToPair(alignmentRegion -> new Tuple2<>(new Tuple2<>(alignmentRegion.assemblyId, alignmentRegion.contigId), alignmentRegion))
+                .groupByKey();
+    }
+
+    // TODO: 12/15/16 test
+    static JavaPairRDD<Tuple2<String, String>, byte[]> loadContigSequence(final JavaSparkContext ctx,
+                                                                          final String pathToInputAssemblies) {
+        return loadContigsCollectionKeyedByAssemblyId(ctx, pathToInputAssemblies)
+                .flatMapToPair(assemblyIdAndContigsCollection -> {
+                    final String assemblyId = assemblyIdAndContigsCollection._1;
+                    final ContigsCollection contigsCollection = assemblyIdAndContigsCollection._2;
+                    return contigsCollection.getContents().stream()
+                            .map(pair -> new Tuple2<>(new Tuple2<>(assemblyId, pair._1.toString()), pair._2.toString().getBytes()))
+                            .collect(Collectors.toList());
+                });
+    }
+
+    // TODO: 11/26/16 assuming 'I'/'D' cannot be the beginning or last CIGAR operator AND the following operator cannot be a clipping
+    /**
```

Comment:
While reviewing some of the calls, there was a region that seem to invalidate this assumption, the particular region is `chr21:44,372,670-44,372,939` in the benchmarking BAM we are using.